lib/rspec/support/caller_filter.rb

```diff
@@ -73,10 +73,24 @@ def self.first_non_rspec_line(skip_frames=3, increment=5)
         end
       end
     else
+      CallerFinder = Class.new(StandardError)
+      class LazyBacktrace
+        def initialize(exception)
+          @exception = exception
+        end
+
+        def to_s
+          @exception.backtrace.find { |line| line !~ IGNORE_REGEX }
+        end
+      end
       # Earlier rubies do not support the two argument form of `caller`. This
       # fallback is logically the same, but slower.
       def self.first_non_rspec_line(*)
-        caller.find { |line| line !~ IGNORE_REGEX }
+        begin
+          raise CallerFinder
+        rescue CallerFinder => error
+          return LazyBacktrace.new(error)
+        end
```

Comment:
As discussed in rspec/rspec-mocks#937, `raise` is modestly _slower_ than `caller` when the backtrace is actually used.  So I don't think we want to always use `raise`...instead, it should be something the caller of `CallerFilter` opts in to due to knowing that the backtrace may not be used.  For situations where we know the backtrace may be used it should use the mechanism it's using now.

WDYT of `RSpec::CallerFilter.first_non_rspec_line` vs `RSpec::LazyCallerFilter.first_non_rspec_line`?